src/discord_integration.py

```diff
@@ -123,7 +123,7 @@ def get_messages(channel_id: int, limit: int = 100) -> list:
 
     # Reverse the list of messages
     new_list.reverse()
-    return new_list
+    return {channel_id: new_list}
```

Comment:
Could you explain why adding a key that corresponds to the `channel_id` and accessing that with `.get()` on the `dict` prevents the "No messages found" from being displayed?

Reply:
It's the simplest way I could think of to make sure that we are updating the view only with the messages from the current channel. Maybe I haven't explained it well enough before, but what it happens is something like this:

1. `discord_integration.get_messages()` gets called with the old channel ID. It takes a while to make the network request, and meanwhile...
2. The user changes the current channel and the program sets the view to "No messages in this conversation yet!"
3. `discord_integration.get_messages()` finishes doing its work and updates the view with the old channel messages a second after the user has changed the channel.
4. After a while, the timer calls `discord_integration.get_messages()` again and updates the view with the messages from the new channel.

What this pull request does is avoid updating the view with the old messages in step 3. To do this, we need to know which is channel the messages returned by discord_integration.get_messages() are from, that's why I add channel_id to the returned value. I guess I could instead return a tuple like `(channeld_id, new_list)`, but this works well enough.